prow/repoowners/repoowners.go

```diff
@@ -241,10 +266,22 @@ func (c *Client) LoadRepoOwners(org, repo, base string) (RepoOwner, error) {
 		}
 		entry.owners, err = loadOwnersFrom(gitRepo.Dir, mdYaml, entry.aliases, dirBlacklist, log)
 		if err != nil {
-			return nil, fmt.Errorf("failed to load RepoOwners for %s: %v", fullName, err)
+			return fmt.Errorf("failed to load RepoOwners for %s: %v", fullName, err)
 		}
 		entry.sha = sha
 		c.cache[fullName] = entry
+		return nil
+	}
+
+	if !ok || entry.sha != sha || entry.owners == nil || entry.owners.enableMDYAML != mdYaml {
+		if err := syncEntry(ok &&
+			entry.sha != "" &&
+			entry.sha != sha &&
+			entry.owners != nil &&
+			entry.owners.enableMDYAML == mdYaml &&
+			entry.aliases != nil); err != nil {
```

Comment:
This pattern is strange. Can you DRY out these conditionals?

Reply:
fixed